src/irmin-pack/unix/file_manager.ml

```diff
@@ -797,4 +797,72 @@ struct
     let chunk_start_idx = pl.chunk_start_idx in
     let chunk_num = pl.chunk_num in
     cleanup ~root ~generation ~chunk_start_idx ~chunk_num
+
+  let create_one_commit_store config ~generation ~new_store_root
+      latest_gc_target =
+    let open Result_syntax in
+    let root = Irmin_pack.Conf.root config in
+    let commit_key, latest_gc_target_offset, suffix_start_offset =
+      match Pack_key.inspect latest_gc_target with
+      | Direct { offset; length; _ } as commit_key ->
+          let suffix_start_offset =
+            Int63.Syntax.(offset + Int63.of_int length)
+          in
+          (commit_key, offset, suffix_start_offset)
+      | Indexed _ ->
+          (* The caller of this function lifted the key to a direct one. *)
+          assert false
+    in
```

Comment:
Instead of doing this calculation here, I would consider changing `create_one_commit_store` to take new parameters of `latest_gc_target_offset` and `suffix_start_offset`, and removing `latest_gc_target`.

1. Treat this as the replacement for `swap` in the snapshot export case
1. Reinforce the usage of this function for after a GC finishes
2. Remove this duplicated logic since the values are already calculated as a part of GC. You could update `finalise_without_swap` to return the values you need in this case. The only thing left then is `commit_key` but if you pass `t` instead of `config` you could load the key that you need based on `latest_gc_target_offset`.

Reply:
I am not sure what you mean by 1. - I moved the call to fm.create_one_commit_store inside the gc module. (I initially didn't do it because I have to pass more arguments to the new Gc.finalise_and.. function, which also is now in charge of more than just finalising the gc). 
I think this change satisfies point 2 as well?

For 3, I removed the duplicated logic. However, I need to pass `config` for opening the new index. `config` is not part of type t for neither `File_manager` or `Gc`. I think its ok to pass on `config` though, its an extra signal that the function opens a new store. Did I miss something?

Lastly for the `commit_key`: the File manager does not access to the commit store in order to load the key. We could load it in the Gc module, but I'm not sure I see why - the commit_key is an argument of the function `create_one_commit_store` we can just pass it along.

Reply:
Ah, sorry for 1. Definitely not enough words. :sweat_smile: I was trying to say that if you change the parameters for this function it starts to look like / follow the pattern `swap` for normal GC but for this new "one commit store" case. I wasn't actually thinking about calling the file manager code from the finalise function, but it works! (my thought was to return the values you needed from `finalise_without_swap`) I'm happy to have the removed duplication.

For `commit_key` my main concern is that `assert false`. Do we know this key is necessarily already direct?

Reply:
> I wasn't actually thinking about calling the file manager code from the finalise function, but it works!

I changed my mind about this, I think it really is not great to have comments explaining that `new_store_root = new_files_path` etc, as you point out in a comment below. Instead these constraints are clearer if we do the call to `gc.v` and `fm.create_` from the same function.

> For commit_key my main concern is that assert false. Do we know this key is necessarily already direct?

Good catch, thanks !

Reply:
Latest diff looks good! :raised_hands: